app/src/main/java/com/kickstarter/features/search/viewmodel/SearchAndFilterViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package com.kickstarter.features.search.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.kickstarter.libs.Environment
+import com.kickstarter.models.Project
+import com.kickstarter.services.DiscoveryParams
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+data class SearchUIState(
+    val isLoading: Boolean = false,
+    val isErrored: Boolean = false,
+    val popularProjectsList: List<Project> = emptyList(), // TODO MBL-2135 popular & search lists could be potentially unified
+    val searchList: List<Project> = emptyList()
+)
+
+class SearchAndFilterViewModel(
+    private val environment: Environment,
+    private val dispatcher: CoroutineDispatcher,
+) : ViewModel() {
+
+    private val apolloClient = requireNotNull(environment.apolloClientV2())
+
+    private val _searchUIState = MutableStateFlow(SearchUIState())
+    val searchUIState: StateFlow<SearchUIState>
+        get() = _searchUIState
+            .asStateFlow()
+            .stateIn(
+                scope = viewModelScope,
+                started = SharingStarted.WhileSubscribed(),
+                initialValue = SearchUIState()
+            )
+
+    // Popular projects sorting selection
+    private val popularDiscoveryParam = DiscoveryParams.builder().sort(DiscoveryParams.Sort.POPULAR).build()
```

Comment:
Was originally gonna create a new Data class to handle the user selection around filtering/searching but taking a look at analytic events and Reftags they are too tied to `DiscoveryParams` therefore will extend `DiscoveryParams` class as needed for this project